{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bin genomes using aviary workflow"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### First symlink reads from samples to use for differential coverage binning to new folder\n",
    "##### Samples are grouped based on conditioning tank and treatment, which showed most consistency in sample replicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Tank MIS5, 2Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221010_metagenome_data_trimmed\"\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE2451,SE2452,SE2453,SE2454,SE2455,SE2457,SE2473,SE2474,SE2475,SE2476,SE2477,SE2496,SE2497,SE2498,SE2499,SE2500,SE2536,SE2537,SE2538,SE2540,SE2541,SE2542}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/MIS5_2ML_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Tank MIS7, 2Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221010_metagenome_data_trimmed\"\n",
    "\n",
    "mkdir /srv/projects/microbial_inducers/data/MIS7_2ML_to_bin\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE2458,SE2459,SE2460,SE2461,SE2463,SE2478,SE2479,SE2480,SE2481,SE2482,SE2484,SE2502,SE2503,SE2505,SE2506,SE2507,SE2543,SE2544,SE2545,SE2546,SE2548}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/MIS7_2ML_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Tank MIS8, 2Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221010_metagenome_data_trimmed\"\n",
    "\n",
    "mkdir /srv/projects/microbial_inducers/data/MIS8_2ML_to_bin\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE2464,SE2465,SE2468,SE2486,SE2487,SE2488,SE2489,SE2490,SE2491,SE2508,SE2509,SE2510,SE2511,SE2512,SE2513,SE2514,SE2549,SE2550,SE2551,SE2552,SE2553,SE2554}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/MIS8_2ML_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MIS5C assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in bactches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS5 (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS5_2ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE2451,SE2454,SE2457,SE2473,SE2477,SE2496}_scaffolds.fasta); do\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS5 (batch 2)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS5_2ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "                        \n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE2498,SE2500,SE2536,SE2537,SE2540}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MIS7C assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in bactches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS7 (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS7_2ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE2458,SE2460,SE2461,SE2478,SE2479,SE2484}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS7 (batch 2)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS7_2ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE2503,SE2506,SE2507,SE2543,SE2544,SE2548}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MIS8C assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in batches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS8 (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS8_2ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE2464,SE2468,SE2485,SE2488,SE2490}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS8 (batch 2)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS8_2ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE2508,SE2510,SE2513,SE2550,SE2553}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
