{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Bin genomes using aviary workflow"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### First symlink reads from samples to use for differential coverage binning to new folder\n",
    "##### Samples are grouped based on conditioning tank and treatment, which showed most consistency in sample replicates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Tank MIS5, 1Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221206_metagenome_data_2_trimmed\"\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE6768,SE6769,SE6770,SE6771,SE6772,SE6787,SE6788,SE6789,SE6790,SE6791,SE6792,SE6797,SE6798,SE6799,SE6800,SE6801,SE6802,SE6803,SE6832,SE6833,SE6834}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/MIS5_1ML_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Tank MIS7, 1Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221206_metagenome_data_2_trimmed\"\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE6773,SE6774,SE6775,SE6776,SE6777,SE6793,SE6794,SE6795,SE6796,SE6804,SE6805,SE6806,SE6807,SE6808,SE6809,SE6810,SE6820,SE6821,SE6822,SE6835,SE6836,SE6837,SE6838,SE6839,SE6840}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/MIS7_1ML_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Tank MIS8, 1Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221206_metagenome_data_2_trimmed\"\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE6778,SE6779,SE6780,SE6781,SE6782,SE6783,SE6784,SE6811,SE6812,SE6813,SE6814,SE6815,SE6816,SE6817,SE6823,SE6824,SE6825,SE6826,SE6827,SE6828,SE6829,SE6841,SE6842,SE6843,SE6844,SE6845,SE6846,SE6847}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/MIS8_1ML_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Dark 2Month light conditioned\n",
    "\n",
    "READS_DIR=\"/srv/projects/microbial_inducers/data/20221206_metagenome_data_2_trimmed\"\n",
    "\n",
    "for FILE in $(ls $READS_DIR/{SE6848,SE6849,SE6850,SE6851,SE6852,SE6853,SE6854,SE6855,SE6856,SE6857,SE6858,SE6859,SE6860,SE6861,SE6862,SE6863,SE6864,SE6865,SE6866,SE6867}*.fastq.gz); do\n",
    "    echo $FILE\n",
    "    ln -s $FILE /srv/projects/microbial_inducers/data/dark_to_bin\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MIS5C assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in batches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS5 (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS5_1ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6768,SE6771,SE6788,SE6792,SE6798}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS5 (batch 2)\n",
    "\n",
    "# load programs\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS5_1ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "                                \n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6801,SE6803,SE6833,SE6834}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MIS7C assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in batches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS7 (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS7_1ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6773,SE6776,SE6793,SE6805,SE6808,SE6810}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS7 (batch 2)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS7_1ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6820,SE6822,SE6836,SE6838,SE6840}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## MIS8C assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in bactches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS8 (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS8_1ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6778,SE6783,SE6784,SE6811,SE6813,SE6816}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all MIS8 (batch 2)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/MIS8_1ML_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6825,SE6828,SE6829,SE6842,SE6844,SE6847}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Dark assemblies"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Running in batches as assemblies finish"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all dark (batch 1)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/dark_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6850,SE6851,SE6853,SE6856,SE6858}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#!/bin/bash\n",
    "\n",
    "# Bin genomes using aviary pipeline\n",
    "\n",
    "# binning all dark (batch 2)\n",
    "\n",
    "# load software\n",
    "module load miniconda3/1.1\n",
    "conda activate aviary_0.5.0\n",
    "\n",
    "# Location of short read files\n",
    "SHORT_READ_DIR=\"/srv/projects/microbial_inducers/data/dark_to_bin\"\n",
    "# Location of scaffolds/assemblies. Should end in '.fasta'\n",
    "SCAFFOLD_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/03_assembly/all_scaffolds\"\n",
    "# Location to output results from aviary\n",
    "AVIARY_DIR=\"/srv/projects/microbial_inducers/analysis/biofilm_metagenome/04_binnning\"\n",
    "\n",
    "#mkdir -p $AVIARY_DIR\n",
    "\n",
    "# Max number of threads to use\n",
    "THREADS=30\n",
    "PPTHREADS=30 \n",
    "# Max memory to allow for bin recovery\n",
    "MEMORY_RECOVER=300\n",
    "\n",
    "# Location to store conda environments (don't change)\n",
    "export CONDA_ENV_PATH=~/.conda/envs/\n",
    "# Location of GTDBTK database (don't change)\n",
    "export GTDBTK_DATA_PATH=/srv/db/gtdbtk/official/release207\n",
    "# Location of EnrichM database (ignore/don't change)\n",
    "export ENRICHM_DB=\"\"\n",
    "# Location of EggNog database (ignore/don't change)\n",
    "export EGGNOG_DATA_DIR=\"\"\n",
    "\n",
    "for FILE in $(ls $SCAFFOLD_DIR/{SE6860,SE6863,SE6864,SE6867}_scaffolds.fasta); do\n",
    "#get variables\n",
    "    SAMPLE=$(basename $FILE | cut -d \"_\" -f 1)\n",
    "    echo \"procesessing scaffold file $FILE\"\n",
    "    echo \"sample name is $SAMPLE\"\n",
    "# run aviary\n",
    "    aviary recover \\\n",
    "    --assembly $FILE \\\n",
    "    --paired_reads_1 $SHORT_READ_DIR/*R1.fastq.gz \\\n",
    "    --paired_reads_2 $SHORT_READ_DIR/*R2.fastq.gz \\\n",
    "    --max_threads $THREADS \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --max_memory $MEMORY_RECOVER \\\n",
    "    --pplacer_threads $THREADS \\\n",
    "    --n_cores $THREADS \\\n",
    "    --output $AVIARY_DIR/$SAMPLE \\\n",
    "    --workflow recover_mags \\\n",
    "    --semibin-model ocean \\\n",
    "    --conda-frontend \"mamba\" 2>&1 | tee $AVIARY_DIR/logs/${SAMPLE}_aviary.log\n",
    "done\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
